docs/_includes/api-options.adoc

```diff
@@ -0,0 +1,149 @@
+//= Ruby API Options
+
+// ref issue 562.
+
+.Ruby API options
+[cols="<15m,<35,<20,<20,<25"]
+|====
+|Name |Description |Default value |Allowed values |CLI equivalent
+
+|attributes
+|Passes map of document attributes.
+|_Empty_
+|Select from <<builtin-attributes>>, or user-defined.
+|`-a`, `--attributes`
+
+|backend
+|Output file format
+|html5
+|html5, docbook5, docbook45, manpage
+|`-b`, `--backend`
+
+|base_dir
+|Base directory containing the document and resources.
+|The directory containing the source file, or the working directory if the source is read from a stream.
+|_path_
+|`-B`, `--base-dir`
+
+|catalog_assets
+|?
+|
+|
+|_none_
+
+|compact
+|Not used
+|false
+|true or false
+|_none_
+
+|converter
+|?
+|
+|
+|_none_
+
+|doctype
+|Document type
+|article
+|article, book, manpage, inline
+|`-d`, `--doctype`
+
+|eruby
+|Specifies the eRuby implementation to use for executing the custom converter templates written in ERB.
+|erb
+|erb, erubis
+|`-e`, `--eruby`
+
+|extensions
+|List of file extensions for asciidoc source files.
+|ad, adoc, asciidoc 
+|
+|_none_
+
+|extensions_registry
+|?
+|
+|
+|_none_
+
+|header_footer
+|If true, suppress the document header and footer in the output.
+|false (opposite to CLI)
+|true or false
+|`-s`, `--no-header-footer`
+
+|mkdirs
+|If true, allows {adr} to create output directory if necessary.
+|false
+|true, false
+|_none_
+
+|parse
+|?
+|
+|
+|_none_
+
+|safe
+|Safe mode.
+|SECURE
+|See CLI
+|`-S`, `--safe-mode`
+
+|source_map
+|?
+|
+|
+|_none_
+
+|template_cache
+|?
+|
+|
+|_none_
+
+|template_dir
+|Specifies a directory of Tilt-compatible templates to be used instead of the default built-in templates.
+|_none_
+|_path_
+|`-T`, `--template-dir`
+
+|template_engine
+|Template engine to use for the custom converter templates.
+The gem with the same name as the engine will be loaded automatically.
+This name is also used to build the full path to the custom converter templates.
+|_none_
+
+Auto-detected based on the file extension of the custom converter templates found.
+|_name_
+|`-E`, `--template-engine`
+
+|template_engine_options
+|?
```

Comment:
The template_engine_options are low-level options passed to the template engine. This option borderlines on being internal and is recommended only for advanced users.

(You can see an example in the Bespoke.js converter at https://github.com/asciidoctor/asciidoctor-bespoke/blob/v1.0.0.alpha.1/lib/asciidoctor-bespoke/converter.rb#L24-L28).